code/game/machinery/alarm.dm

```diff
@@ -976,13 +989,13 @@ var/global/list/air_alarms = list()
 		else
 			max_temperature = temperature_threshold.max_1() - T0C
 			min_temperature = temperature_threshold.min_1() - T0C
-		var/input_temperature = input("What temperature (in C) would you like the system to target? (Capped between [min_temperature]C and [max_temperature]C).\n\nNote that the cooling unit in this air alarm can not go below [MIN_TEMPERATURE]C or above [MAX_TEMPERATURE]C by itself. ", "Thermostat Controls") as num|null
+		var/input_temperature = input("What temperature (in C) would you like the system to target? (Capped between [min_temperature]C and [max_temperature]C).\n\nNote that the cooling unit in this air alarm can not go below [MIN_TEMPERATURE - T0C]C or above [MAX_TEMPERATURE - T0C]C by itself. ", "Thermostat Controls") as num|null
 		if(input_temperature==null)
 			return 1
-		if(!input_temperature || input_temperature >= max_temperature || input_temperature <= min_temperature)
+		if(input_temperature > max_temperature || input_temperature < min_temperature)
```

Comment:
Could also just allow values outside the range but clamp the result. Atmos computers work like this.

Reply:
Agreed.